.github/workflows/codeql.yml

```diff
@@ -0,0 +1,154 @@
+#   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-  .-.-.   .-.-.   .-.-
+#  / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ / / \ \ / / \ \ / / \
+# `-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-'   `-`-'   `-`-'   `-`-'
+#
+#                       CodeQL
+#
+#  This workflow replaces the GitHub CodeQL extension to support fork PRs.
+#  The extension doesn't trigger on fork PRs due to security restrictions.
+#  This workflow uses the pull_request event which works for all PRs.
+#
+#   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-.   .-.-  .-.-.   .-.-.   .-.-
+#  / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ \ / / \ / / \ \ / / \ \ / / \
+# `-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-`-'   `-'   `-`-'   `-`-'
+name: CodeQL
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+  pull_request:
+    branches:
+      - main
+      - develop
+  merge_group:
+    # Required for GitHub merge queue
+    branches:
+      - main
+      - develop
+  schedule:
+    # Run weekly security scans every Monday at midnight UTC
+    - cron: '0 0 * * 1'
+
+# Cancel in-progress runs when a new commit is pushed
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Comment:
I think this will only cancel in-progress runs in a same PR.

I'm guessing it's ok.